src/Analyzer/FileVisitor.php

```diff
@@ -45,12 +47,24 @@ public function onClassAnalyzed(callable $callable): void
         $this->callable = $callable;
     }
 
+    public function getClassDescriptions(): array
+    {
+        return $this->classDescriptions;
+    }
+
+    public function clearParsedClassDescriptions(): void
+    {
+        $this->classDescriptions = [];
+    }
+
     public function leaveNode(Node $node): void
     {
         if ($node instanceof Node\Stmt\Class_) {
             $classDescription = $this->classDescriptionBuilder->get();
 
-            \call_user_func($this->callable, $classDescription);
+            $this->classDescriptions[] = $classDescription;
```

Comment:
instead of relying on a callback we accumulate class description here

Reply:
Make sense this approach